packages/core/src/interpreter.ts

```diff
@@ -844,17 +849,14 @@ export class Interpreter<
             : serviceCreator;
 
           if (isPromiseLike(source)) {
-            this.state.children[id] = this.spawnPromise(
```

Comment:
q: so those were setting the child on the not-yet-updated `this.state`?

Reply:
I don't remember the specifics, but these did appear on the updated `this.state`

Reply:
ok, so I've jumped into the code to understand this a little bit better and this is what I've learns - `spawn` is not handled by this part of the code at all, this is processing of `start` actions (so basically for starting invokes / activities) and that's why this has not been working. This is IMHO a very strong candidate for the refactor in v5 because invoke/spawn should IMHO behave in the very same way. The only difference should be that invokes are static and bound to state whereas spawned actors are dynamic.

However, right now they differ because invokes are more lazy and spawns are eager. For example if an invoke gets "immediately" cancelled within a single macrotask then it is not even started (according to the SCXML spec `statesToInvoke` are only processed after the `internalQueue` gets empty), but `spawn` does not behave in this way as it's eager (handled right away, rather than deferred to the end of the current macrotask).

Reply:
> but `spawn` does not behave in this way as it's eager (handled right away, rather than deferred to the end of the current macrotask).

This behavior is fixed in the `next` branch already. Spawned actors will behave exactly the same way.

Reply:
That's great to hear 🚀